### PR TITLE
feat(pml-launcher): Add route and scaffold new forms under testflag

### PR DESCRIPTION
**Pull request by jaredvu**
Screenshot 2024-08-21 at 12 15 07 PM

Screenshot 2024-08-21 at 12 16 50 PM

Launch Market page
- Set-up app routes for `/launch-market` url
- Scaffold NewMarketPreview and NewMarketSelection forms

---

## Views

* New `<LaunchMarket>`
  * add new page
  * rough scaffold of components on page

* New `<NewMarketPreviewStep>` and `<NewMarketSelectionStep>`
  * Rough scaffold of brand new forms in the `v7` directory (protocol support version)

* `<HeaderDesktop>`
  * add `Launch Market` item behind testFlag

* `<App>`
  * Add new route behind testflag and redirect `/markets/new` -> `/launch-market`

* `<NewMarketForm>`
  * add new components under `pml` flag
  * add hook to get receipt items since receipts are shared on two forms

## Components

* `<NavigationMenu>`
  * `white-space: nowrap` for nav items

## Constants/Types

* `constants/routes`
  * add new route

## Functions

* `lib/testFlags`
  * add `pml` testflag

## Hooks

* `hooks/useLaunchableMarkets`
  * Add dummy query (only queries hardcoded staging url) to get market map data
  * returns filtered list (referencing existing marketIds)

---

### CODE REVIEW

File: src/hooks/useLaunchableMarkets.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      return [];
+    }
+
+    return Object.entries(marketMap)
+      .map(([id, data]) => ({
+        id,
+        ...data,
+      }))
+      .filter(({ id }) => {
+        if (marketIds.includes(id)) {
```
Comment (tyleroooo) [inline src/hooks/useLaunchableMarkets.ts:81]: `return !marketIds.includes(id)` also works

File: src/views/forms/NewMarketForm/index.tsx [hunk 2]
```diff
@@ -35,6 +46,87 @@ export const NewMarketForm = () => {
     return Math.abs(p - 3);
   }, [assetToAdd]);
 
+  const receiptItems = useMemo(() => {
+    return [
+      {
+        key: 'deposit-apr',
+        label: 'Deposit APR (30d)',
+        value: (
+          <Output type={OutputType.Percent} value={0.3256} fractionDigits={PERCENT_DECIMALS} />
```
Comment (tyleroooo) [inline src/views/forms/NewMarketForm/index.tsx:55]: is PERCENT_DECIMALS necessary?

File: src/views/forms/NewMarketForm/v7/NewMarketPreviewStep.tsx [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+import { useAppDispatch } from '@/state/appTypes';
+import { openDialog } from '@/state/dialogs';
+
+import { log } from '@/lib/telemetry';
+
+type NewMarketPreviewStepProps = {
+  ticker: string;
+  onBack: () => void;
+  onSuccess: (ticker: string) => void;
+  receiptItems: Parameters<typeof Details>[0]['items'];
```
Comment (tyleroooo) [inline src/views/forms/NewMarketForm/v7/NewMarketPreviewStep.tsx:31]: I think there's a PropType type somewhere that gives you the props which might be easier to read here

**Review by tyleroooo — APPROVED** [review]
seems sane to meeeeeee

File: src/hooks/useLaunchableMarkets.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+export type HydratedMarketMap = MarketMapResponse['market_map']['markets'][string] & { id: string };
+
+export const useMarketMap = () => {
+  const { selectedNetwork } = useSelectedNetwork();
+
+  const launchableMarkets = useQuery({
+    enabled: selectedNetwork === 'dydxprotocol-staging',
+    queryKey: ['launchableMarkets', selectedNetwork],
+    queryFn: async () => {
+      return fetch('https://validator.v4staging.dydx.exchange:1317/slinky/marketmap/v1/marketmap')
```
Comment (tyleroooo) [inline src/hooks/useLaunchableMarkets.ts:56]: do we have a plan for not hardcoding this? I was wondering if we should @JsExport the V4StateManagerConfigs object from abacus which handles a lot of this

Comment (jaredvu) [reply src/hooks/useLaunchableMarkets.ts:56]: Yeah the plan is to add the fetch to v4-client-js and expose from our useDydxClient hook.

**Review by tyleroooo — APPROVED** [review]

**Pull request merged**